package.json

```diff
@@ -22,6 +23,7 @@
     "angular-toarrayfilter": "1.0.3",
     "angular-ui-router": "0.4.3",
     "angular1-ui-bootstrap4": "2.4.22",
+    "auth0-js": "^9.0.2",
```

Comment:
Look at the Auth0 docs I see them mentioning an iframe version called `auth0-lock`, with "Auth0 Lock" having been mentioned in the bug too. Which should we be using? 

The comparison in their docs (https://auth0.com/docs/libraries/when-to-use-lock) plus the bug comments make me think Lock is soon going to be the preferred approach? Or is that not possible with what we need here? (I've not looked at Lock in detail)

Reply:
At Mozilla, we want to use the "hosted lock" as it embodies all the logic of who can login with LDAP, who with github, etc.

Reply:
So it sounds like my question above is relevant? ie should be using the `auth0-lock` library instead?

Reply:
I think one would use the lock widget if they want to customize some aspect of the dialog's appearance and behavior. Also, I think if you use the lock widget, then some features will be blocked. 

From the auth0 docs:

![screen shot 2018-01-24 at 5 51 01 pm](https://user-images.githubusercontent.com/3766511/35361639-0cd321b6-0130-11e8-9745-d33a514af605.png)

The added customatization stuff makes `auth0-lock` a very heavy package heh. Using something like ncdu, I get: 

<img width="540" alt="screen shot 2018-01-24 at 6 04 41 pm" src="https://user-images.githubusercontent.com/3766511/35361930-47356fde-0131-11e8-8a37-4ff94de77510.png">

Reply:
Oh so auth0-lock != "hosted lock" ?

Reply:
Either way, all the better if we don't need it then, given the size :-)

Reply:
The docs are very confusing. I think we are using a Custom UI (not a lock) in order to support things like LDAP logins. We are using our own lock UI which is stored in https://github.com/mozilla-iam/auth0-deploy/blob/master/pages/login.html.

Reply:
Yeah, it's all very confusing.  The "Hosted" lock is one the IAM team implemented and uploaded to `auth.mozilla.auth0.com`, *not* auth0-lock.js.